org-code-javabuilder/lib/src/main/java/org/code/javabuilder/JavaRunner.java

```diff
@@ -93,11 +90,13 @@ private boolean run(
             classLoaderUrls, JavaRunner.class.getClassLoader(), classNames, permissionLevel);
 
     boolean runResult;
-    PerformanceTracker.getInstance().trackUserCodeStart();
+    PerformanceTracker performanceTracker =
+        (PerformanceTracker) JavabuilderContext.getInstance().get(PerformanceTracker.class);
```

Comment:
General comment, and this can definitely be a follow-up - with these singletons being moved into JavabuilderContext, we could also now make them constructor arguments which can be more easily mocked in unit tests (instead of having to mock static objects) - for example here `performanceTracker` can be passed in via the constructor

Reply:
are you thinking `performanceTracker` would be a constructor argument to `JavabuilderContext`?

Reply:
Oh I meant `performanceTracker` could be an instance variable on `JavaRunner` passed in via constructor overloading - 
```
public JavaRunner(...) {
  this(..., JavabuilderContext.get(PerformanceTracker.class), ...);
}
```
So that we way we can pass in a `performanceTracker` mock directly in unit tests rather than having to mock `JavabuilderContext` or the static `PerformanceTracker` class (this example maybe isn't the best since I don't think we currently have a test for `JavaRunner`, but just thinking in general that JavabuilderContext can allow us to use this pattern)

Reply:
gotcha, yeah I think I'll leave as-is but will do a follow-up if there's classes where that pattern makes sense for tests.